File: benchmarks/java/WordFreq.java
import java.nio.file.*;
import java.util.*;
public class WordFreq {
    public static void main(String[] args) throws Exception {
        String text = Files.readString(Path.of("input.txt"));
        HashMap<String, Long> counts = new HashMap<>();
        for (String word : text.split("\\s+")) {
            if (word.isEmpty()) continue;
            counts.merge(word, 1L, Long::sum);
        }
        ArrayList<String> ranked = new ArrayList<>();
        for (Map.Entry<String, Long> entry : counts.entrySet())
            ranked.add(String.format("%8d %s", entry.getValue(), entry.getKey()));
        ranked.sort(Comparator.reverseOrder());
        for (int i = 0; i < 3; i++) System.out.println(ranked.get(i).trim());
        System.out.println(counts.size() + " unique");
    }
}

File: benchmarks/rust/wordfreq.rs
use std::collections::HashMap;
use std::fs;
fn main() {
    let text = fs::read_to_string("input.txt").unwrap();
    let mut counts: HashMap<&str, i64> = HashMap::new();
    for word in text.split_whitespace() { *counts.entry(word).or_insert(0) += 1; }
    let mut ranked: Vec<String> = counts.iter().map(|(w, c)| format!("{:>8} {}", c, w)).collect();
    ranked.sort();
    ranked.reverse();
    for line in ranked.iter().take(3) { println!("{}", line.trim()); }
    println!("{} unique", counts.len());
}
